hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/distributed/IMapRegionCache.java

```diff
@@ -115,8 +117,9 @@ public boolean remove(final Object key) {
 
     public SoftLock tryLock(final Object key, final Object version) {
         long timeout = nextTimestamp(hazelcastInstance) + lockTimeout;
-
-        return (SoftLock) map.executeOnKey(key, new LockEntryProcessor(nextMarkerId(), timeout, version));
+        final ExpiryMarker marker = (ExpiryMarker) map.executeOnKey(key,
+                new LockEntryProcessor(nextMarkerId(), timeout, version));
+        return new MarkerWrapper(marker);
     }
 
     public void unlock(final Object key, SoftLock lock) {
```

Comment:
You need to get the marker from the softlock here. `((MarkerWrapper) lock).getMarker()`.